frosthaven_assistant/lib/services/network/communication.dart

```diff
@@ -42,14 +48,28 @@ class Communication {
     }
   }
 
-  void disconnect() {
+  void disconnectAll() {
     while (_sockets.isNotEmpty) {
       var socket = _sockets.first;
       socket.destroy();
       _sockets.remove(socket);
     }
   }
 
+  void disconnect(Socket socket) {
+    var toDisconnect = _sockets
+        .where((x) =>
+            x.remoteAddress == socket.remoteAddress && x.port == socket.port)
```

Comment:
you may want to put this in try catch block as accessing remoteaddress will throw error throw in case the socket has been closed remotely

Reply:
Yeah, I saw that try/catch block, and didn't quite get it.
Is it happening frequently?
I will update this, but it looks really weird 😄

Reply:
I couldn't check if it works for the case you mentioned, as I couldn't re-produce it, sorry, lack understanding of sockets a bit.
Nevertheless, added check if address and port are retrievable from socket :)

Reply:
I'm not an expert on the tcp sockets, but basically a socket can be closed on one end without the other knowing (say you cut the power on the server machine, no end service message can be sent, or if the client machine network gets interrupted for example, same issue. You can also close a socket on one end on purpose with close(), though not sure why you 'd want that, instead of doing a socket.destroy() which kills it on both ends. Furthermore the server and client sockets work slightly different.

Honestly I believe the remoteAddress throwing an error is a bit of a trap, since it's easy to think it's just another local value, unless you read the documentation.

Reply:
So, what do you think about changes?
Should work with try/catch 😊

Reply:
Just tried now with a windows server and android client.
if after connect I turn wifi off on android device there's an unhandled SocketException (errno 103) on the client side.
Should handle the exception.

same goes for turning off wifi on server side.

Though... it seems this was the case before this MR too. So I might as well just merge it, and see what needs to be done for this case.

Reply:
Hmm... I will check that as well. But yes, disconnection should've handled that properly

Reply:
@Tarmslitaren I've taken a look at that error, and it looks like `network_info_plus` bug of type conversion from Null to String. Null is handled properly in the future but for some reason they forgot to put nullable string type on one variable. At least that is the case for Linux :D